automation/qbft/scenarios/integration_test_type.go

```
@@ -0,0 +1,364 @@
+	"github.com/bloxapp/ssv/protocol/v2/types"
+	"github.com/bloxapp/ssv/storage"
+	"github.com/bloxapp/ssv/storage/basedb"
+	"github.com/bloxapp/ssv/utils/logex"
+	"github.com/bloxapp/ssv/utils/rsaencryption"
+)
+
+// IntegrationTest defines an integration test.
+type IntegrationTest struct {
+	Name              string
+	OperatorIDs       []spectypes.OperatorID
+	InitialInstances  map[spectypes.OperatorID][]*protocolstorage.StoredInstance
+	Duties            map[spectypes.OperatorID][]*spectypes.Duty
+	ExpectedInstances map[spectypes.OperatorID][]*protocolstorage.StoredInstance
+	StartDutyErrors   map[spectypes.OperatorID]error
+}
+
+type scenarioContext struct {
+	ctx         context.Context
+	logger      *zap.Logger
+	nodes       map[spectypes.OperatorID]network.P2PNetwork      // 1 per operator, pass same to each instance
+	nodeKeys    map[spectypes.OperatorID]testing.NodeKeys        // 1 per operator, pass same to each instance
+	stores      map[spectypes.OperatorID]*qbftstorage.QBFTStores // 1 store per operator, pass same store to each instance
+	keyManagers map[spectypes.OperatorID]spectypes.KeyManager    // 1 per operator, pass same to each instance
+	dbs         map[spectypes.OperatorID]basedb.IDb              // 1 per operator, pass same to each instance
+}
+
+func (it *IntegrationTest) bootstrap(ctx context.Context) (*scenarioContext, error) {
+	loggerFactory := func(s string) *zap.Logger {
+		return logex.Build("simulation", zapcore.DebugLevel, nil).With(zap.String("who", s))
+	}
+	logger := loggerFactory(fmt.Sprintf("Bootstrap/%s", it.Name))
+	logger.Info("creating resources")
+
+	types.SetDefaultDomain(spectypes.PrimusTestnet)
+
+	dbs := make(map[spectypes.OperatorID]basedb.IDb)
+	for _, operatorID := range it.OperatorIDs {
+		db, err := storage.GetStorageFactory(basedb.Options{
+			Type:   "badger-memory",
+			Path:   "",
+			Logger: zap.L(),
+		})
+		if err != nil {
+			logger.Panic("could not setup storage", zap.Error(err))
```

let's return an error

---

done @amir-blox